### PR TITLE
REMOVE peer only listeners API

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

REMOVE the peer-only channel implementation

On both sides of the channel, events should 'just work'

### CODE REVIEW

File: lib/channels/src/index.ts [hunk 1]
```diff
@@ -60,9 +58,8 @@ export class Channel {
   }
 
   addPeerListener(eventName: string, listener: Listener) {
-    const peerListener = listener;
-    peerListener.ignorePeer = true;
-    this.addListener(eventName, peerListener);
+    console.warn('channel.addPeerListener is deprecated');
```
Comment (shilman) [inline lib/channels/src/index.ts:61]: Please use `util-deprecate` for consistency

File: lib/channels/src/index.ts [hunk 2]
```diff
@@ -73,10 +70,10 @@ export class Channel {
     }
 
     const handler = () => {
+      this.handleEvent(event);
```
Comment (shilman) [inline lib/channels/src/index.ts:73]: What's the rationale?

**Review by shilman — CHANGES_REQUESTED** [review]
LGTM modulo one question & one small fix

File: lib/channels/src/index.ts (hunk 2 above)
Comment (ndelangen) [reply lib/channels/src/index.ts:73]: Hmm it's been a while.. I did have a reason for this. it was something timing related, or catching an error.

I'll check what happens if I reverse it again.

Comment (ndelangen) [reply lib/channels/src/index.ts]: nothing happens, I'll revert this.

File: lib/channels/src/index.ts (hunk 1 above)
Comment (ndelangen) [reply lib/channels/src/index.ts]: done

**Review by shilman — APPROVED** [review]
Awesome !!

**Pull request merged**